swift/src/Ice/LocalExceptionFactory.swift

```diff
@@ -2,206 +2,59 @@
     }
 
-    static func connectFailedException(_ error: Int32, file: String, line: Int32) -> Error {
-        return ConnectionRefusedException(error: error, file: file, line: line)
-    }
-
-    static func connectionLostException(_ error: Int32, file: String, line: Int32) -> Error {
-        return ConnectionLostException(error: error, file: file, line: line)
-    }
-
-    static func socketException(_ error: Int32, file: String, line: Int32) -> Error {
-        return SocketException(error: error, file: file, line: line)
-    }
-
-    static func syscallException(_ error: Int32, file: String, line: Int32) -> Error {
-        return SyscallException(error: error, file: file, line: line)
-    }
-
-    static func connectionIdleException(_ file: String, line: Int32) -> Error {
-        return ConnectionIdleException(file: file, line: line)
-    }
-
-    static func connectTimeoutException(_ file: String, line: Int32) -> Error {
-        return ConnectTimeoutException(file: file, line: line)
-    }
-
-    static func closeTimeoutException(_ file: String, line: Int32) -> Error {
-        return CloseTimeoutException(file: file, line: line)
-    }
-
-    static func invocationTimeoutException(_ file: String, line: Int32) -> Error {
-        return InvocationTimeoutException(file: file, line: line)
-    }
-
-    static func timeoutException(_ file: String, line: Int32) -> Error {
-        return TimeoutException(file: file, line: line)
-    }
-
-    static func closeConnectionException(_ reason: String, file: String, line: Int32) -> Error {
-        return CloseConnectionException(reason: reason, file: file, line: line)
-    }
-
-    static func connectionManuallyClosedException(_ graceful: Bool, file: String, line: Int32)
+    static func localException(_ typeId: String, message: String, cxxDescription: String, file: String, line: Int32)
         -> Error
     {
-        return ConnectionManuallyClosedException(graceful: graceful, file: file, line: line)
-    }
-
-    static func datagramLimitException(_ reason: String, file: String, line: Int32) -> Error {
-        return DatagramLimitException(reason: reason, file: file, line: line)
-    }
-
-    static func marshalException(_ reason: String, file: String, line: Int32) -> Error {
-        return MarshalException(reason: reason, file: file, line: line)
-    }
-
-    static func protocolException(_ reason: String, file: String, line: Int32) -> Error {
-        return ProtocolException(reason: reason, file: file, line: line)
+        let className = typeId.dropFirst(2).replacingOccurrences(of: "::", with: ".")
+        return if let localExceptionType = NSClassFromString(className) as? LocalException.Type {
+            localExceptionType.init(message: message, cxxDescription: cxxDescription, file: file, line: line)
+        } else {
+            CxxLocalException(typeId: typeId, message: message, cxxDescription: cxxDescription, file: file, line: line)
```

Comment:
Used for C++ local exceptions not mapped to Swift exceptions such as Ice::LocalException.